specs/WebAuthenticationRequested.md

```diff
@@ -0,0 +1,276 @@
+            })
+            .Get(),
+        &m_BasicAuthenticationRequestedToken));
+```
+
+```c#
+webView.CoreWebView2.BasicAuthenticationRequested += delegate (object sender, CoreWebView2BasicAuthenticationRequestedEventArgs args)
+{
+    CoreWebView2Deferral deferral = args.GetDeferral();
+    Credential credential = await ShowCustomLoginUIAsync();
+    args.Response.UserName = credential.UserName;
+    args.Response.Password = credential.Password;
+    deferral.Complete();
+};
+```
+
+## Cancel authentication prompt
+The developer can block the authentication request. In this case, the default login dialog prompt will no longer be shown to the user and the server will respond as if the user clicked cancel.
+
+```cpp
+    CHECK_HRESULT(webview2->add_BasicAuthenticationRequested(
+        Callback<ICoreWebView2BasicAuthenticationRequestedEventHandler>(
+            [this](
+                ICoreWebView2* sender,
+                ICoreWebView2BasicAuthenticationRequestedEventArgs* args)
+            {
+                CHECK_HRESULT(args->put_Cancel(true));
+
+                return S_OK;
+            })
+            .Get(),
+        &m_BasicAuthenticationRequestedToken));
+```
+
+```c#
+webView.CoreWebView2.BasicAuthenticationRequested += delegate (object sender, CoreWebView2BasicAuthenticationRequestedEventArgs args)
+{
+    args.Cancel = true;
+};
+```
+
+## Read authentication challenge string
+Developer can read the authentication challenge string sent by server. Note that if the developer doesn't cancel or provide a response, the default login dialog prompt will be shown to the user.
+
+```cpp
+webview2->add_BasicAuthenticationRequested(
+    Callback<ICoreWebView2BasicAuthenticationRequestedEventHandler>(
+        [this](
+            ICoreWebView2* sender,
+            ICoreWebView2BasicAuthenticationRequestedEventArgs* args)
+        {
+            CHECK_HRESULT(args->get_Challenge(&challenge));
+            if (!ValidateChallenge(challenge.get()))
+            { // Check the challenge string
+                CHECK_HRESULT(args->put_Cancel(true));
+            }
+            return S_OK;
+        })
+        .Get(),
+    &m_BasicAuthenticationRequestedToken));
+```
+
+```c#
+webView.CoreWebView2.BasicAuthenticationRequested += delegate (object sender, CoreWebView2BasicAuthenticationRequestedEventArgs args)
+{
+    if (!ValidateChallenge(args.Challenge))
+    {
+        args.Cancel = true;
+    }
+};
+```
+
+# Remarks
+
+
+# API Notes
+
+
+# API Details
+```idl
+/// WebView2 enables you to host web content using the latest Microsoft Edge
+/// browser and web technology.
+
+[uuid(76eceacb-0462-4d94-ac83-423a6793775e), object, pointer_default(unique)]
+interface ICoreWebView2_4 : ICoreWebView2_3
+{
+  /// ...
+
+  /// Add an event handler for the BasicAuthenticationRequested event.
+  /// BasicAuthenticationRequested event is raised when WebView encounters a Basic HTTP
+  /// Authentication request as described in
```

Comment:
Or NTLM?

Reply:
Please update documentation to note this